building_sim_plugins/building_ignition_plugins/src/slotcar.cpp

```diff
@@ -201,9 +239,53 @@ std::vector<Eigen::Vector3d> SlotcarPlugin::get_obstacle_positions(
   return obstacle_positions;
 }
 
+void SlotcarPlugin::charge_state_cb(const ignition::msgs::Selection& msg)
+{
+  dataPtr->charge_state_cb(msg.name(), msg.selected());
+}
+
+// First element of msg should be the slotcar, and the second should be the payload
+void SlotcarPlugin::item_dispensed_cb(const ignition::msgs::UInt64_V& msg)
+{
+  if (msg.data_size() == 2 && msg.data(0) == _entity)
+  {
+    Entity new_payload = msg.data(1);
+    this->_payloads.insert(new_payload);
+  }
+}
+
+void SlotcarPlugin::item_ingested_cb(const ignition::msgs::Entity& msg)
+{
+  if (msg.IsInitialized() && _payloads.find(msg.id()) != _payloads.end())
+  {
+    _payloads.erase(msg.id());
+  }
```

Comment:
Nit, we are doing two lookups in the `unordered_set` for the same `msg.id()`, I'd probably just create an iterator in the if statement, check it and erase it to do a single lookup.